kits/js/ResultWidget.js

```diff
@@ -51,6 +51,40 @@ jT.ItemListWidget.prototype = {
       }
     }
   },
+  renderLinks: function (doc) {
+    var baseUrl = this.getBaseUrl(doc),
+        item = {};
+
+    // Check if external references are provided and prepare and show them.
+    if (doc.content == null) {
+      item.link = baseUrl + doc.s_uuid;
+      item.href = item.link + "/study";
+      item.href_title = "Study";
+      item.href_target = doc.s_uuid;
+      item.footer = 
+        '<a href="' + baseUrl + doc.s_uuid + '" title="Substance" target="' + doc.s_uuid + '">Material</a>' +
+        '<a href="' + baseUrl + doc.s_uuid + '/structure" title="Composition" target="' + doc.s_uuid + '">Composition</a>' +
+        '<a href="' + baseUrl + doc.s_uuid + '/study" title="Study" target="' + doc.s_uuid + '">Studies</a>';
+      item.composition = this.renderComposition(doc, 
+        '<a href="' + baseUrl + doc.s_uuid + '/structure" title="Composition" target="' + doc.s_uuid + '">&hellip;</a>').join("<br/>");
+        ;
+    } else {
+      item.href_title = this.tagDbs[doc.dbtag_hss] && this.tagDbs[doc.dbtag_hss].title || "External database";
+      item.href_target = "external";
+      item.composition = this.renderComposition(doc);
+      
+      for (var i = 0; i < doc.content.length; ++i) {
+        if (!doc.content[i].match(/^https?:\/\//))
+          continue;
+        if (!item.link)
+          item.link = doc.content[i];
+
+        item.footer += '<a href="' + doc.content[i] + '" target="external">External database</a> ';
```

Comment:
should be `item.footer = ` , otherwise see the `undefined` part 

![image](https://user-images.githubusercontent.com/211190/76142573-a727ce80-6077-11ea-8bcd-06098e99211c.png)

Reply:
Also the `item.href_title`  is not used  but should be (the same line 82)

Reply:
Good catch for `item.footer`. It is a bit unclear for me how should an array of `doc.content` values be treated. As you can see, currently I'm listing them, one after another.

I've made a bunch of fixes regarding `link`, `href` and other involved properties.